cli/src/commands/alpha/take_more_labels.rs

```diff
@@ -0,0 +1,260 @@
+                ErrorKind::Client("Getting batch of target comments failed.".into())
+            })?;
+            statistics.add_target_comments(target_comment_batch.len());
+
+            // If we care about the target comment, return a lookup of the comment id and existing
+            // target labels
+            let mut target_labellings_to_update =
+                make_labellings_lookup(target_comment_batch, &only_for_target_label);
+
+            // If we don't care about any comments in this batch, get next one
+            if target_labellings_to_update.is_empty() {
+                continue;
+            }
+
+            // Fetch source labellings for our given ids
+            let source_annotated_comments = client
+                .get_labellings(
+                    &source_dataset.full_name(),
+                    target_labellings_to_update.keys(),
+                )
+                .chain_err(|| {
+                    ErrorKind::Client("Getting batch of source labellings failed.".into())
+                })?;
+
+            // For each set of source annotations we need to merge in
+            for source_annotated_comment in source_annotated_comments.into_iter() {
+                let AnnotatedComment {
+                    comment: Comment { uid, .. },
+                    labelling: source_labelling,
+                    ..
+                } = source_annotated_comment;
+                let source_assigned = source_labelling.unwrap_or_default().assigned;
+                if source_assigned.is_empty() {
+                    continue;
+                }
+                // Get the matching dismissed annotations we already have
+                let Labelling {
+                    assigned: target_assigned,
+                    dismissed,
+                    ..
+                } = target_labellings_to_update
+                    .remove(&uid)
+                    .expect("Tried to make a new labelling for a comment we don't need to update.");
+
+                // Merge assigned labels from the two datasets
+                let assigned = merge_assigned_labels(
+                    target_assigned,
+                    source_assigned,
+                    &transferred_label_parent,
+                );
+                let new_labelling = NewLabelling {
+                    assigned,
+                    dismissed,
+                };
+
+                // Push the new merged annotations up to the API
+                client
+                    .update_labelling(
+                        &target_dataset.full_name(),
+                        &uid,
+                        Some(&new_labelling),
+                        None,
+                    )
+                    .chain_err(|| ErrorKind::Client("Updating labelling failed.".into()))?;
+                statistics.add_labellings_updated(1);
+            }
+
+            let _progress = make_progress()?;
```

Comment:
Maybe there should just be a single top-level progress bar across all source ids?